includes/sanitizers/class-amp-style-sanitizer.php

```diff
@@ -549,22 +607,47 @@ private function process_link_element( DOMElement $element ) {
 	 * }
 	 * @return array Processed stylesheet parts.
 	 */
-	private function process_stylesheet( $stylesheet, $node, $options = array() ) {
-		$cache_impacting_options = wp_array_slice_assoc(
-			$options,
-			array( 'property_whitelist', 'property_blacklist', 'stylesheet_url', 'allowed_at_rules' )
+	private function process_stylesheet( $stylesheet, $options = array() ) {
+		$parsed      = null;
+		$cache_key   = null;
+		$cache_group = 'amp-parsed-stylesheet-v5';
+
+		$cache_impacting_options = array_merge(
+			wp_array_slice_assoc(
+				$options,
+				array( 'property_whitelist', 'property_blacklist', 'stylesheet_url', 'allowed_at_rules' )
+			),
+			wp_array_slice_assoc(
+				$this->args,
+				array( 'should_locate_sources', 'parsed_cache_variant' )
+			)
 		);
 
-		$cache_key = md5( $stylesheet . serialize( $cache_impacting_options ) ); // phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions.serialize_serialize
+		$cache_key = md5( $stylesheet . wp_json_encode( $cache_impacting_options ) );
```

Comment:
It's nice how using `wp_json_encode()` removes the need to ignore a `phpcs` violation.